modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCRPInitiatedLogoutTestCase.java

```diff
@@ -0,0 +1,288 @@
+import org.wso2.identity.integration.test.rest.api.user.common.model.Email;
+import org.wso2.identity.integration.test.rest.api.user.common.model.Name;
+import org.wso2.identity.integration.test.rest.api.user.common.model.UserObject;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This test class tests the OIDC RP-Initiated logout flows
+ */
+public class OIDCRPInitiatedLogoutTestCase extends OIDCAbstractIntegrationTest {
+
+    protected UserObject user;
+    protected String idToken;
+    protected String sessionDataKeyConsent;
+    protected String sessionDataKey;
+    protected AuthorizationCode authorizationCode;
+    CookieStore cookieStore = new BasicCookieStore();
+    protected Lookup<CookieSpecProvider> cookieSpecRegistry;
+    protected RequestConfig requestConfig;
+    protected HttpClient client;
+    protected List<NameValuePair> consentParameters = new ArrayList<>();
+    OIDCApplication playgroundApp;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+
+        initUser();
+        createUser(user);
+        userInfo.setUserName(user.getUserName());
+        userInfo.setPassword(user.getPassword());
+
+        playgroundApp = initApplication();
+        createApplication(playgroundApp);
+
+        cookieSpecRegistry = RegistryBuilder.<CookieSpecProvider>create()
+                .register(CookieSpecs.DEFAULT, new RFC6265CookieSpecProvider())
+                .build();
+        requestConfig = RequestConfig.custom()
+                .setCookieSpec(CookieSpecs.DEFAULT)
+                .build();
+        client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore)
+                .setDefaultCookieSpecRegistry(cookieSpecRegistry)
+                .setDefaultRequestConfig(requestConfig)
+                .build();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        deleteUser(user);
+        deleteApplication(playgroundApp);
+        clear();
+    }
+
+    @AfterMethod
+    public void clearVariables() {
+
+        sessionDataKey = null;
+        sessionDataKeyConsent = null;
+    }
+
+    @Test(groups = "wso2.is", description = "Test RP-initiated logout with client_id parameter")
+    public void testOIDCLogoutWithClientId() throws Exception {
+
+        testInitiateOIDCRequest(playgroundApp, client);
+        testOIDCLogin(playgroundApp, true);
+        testOIDCConsentApproval(playgroundApp);
+        testOIDCLogout(new BasicNameValuePair("client_id", playgroundApp.getClientId()));
+    }
+
+    @Test(groups = "wso2.is", description = "Test RP-initiated logout with id_token_hint parameter")
+    public void testOIDCLogoutWithIdTokenHint() throws Exception {
```

Comment:
Why isnt testOIDCConsentApproval here? is it becuase consent is already approved from the previous test? if that is the case add a dependOn for the testOIDCLogoutWithClientId in here

Reply:
Yes, consent is already approved from the clientId test. added a dependsOn with [95e6b7e](https://github.com/wso2/product-is/pull/16746/commits/95e6b7e0ad6fe1815a1ee840b21b43a0e81a0a55)